src/libs/Network/index.js

```diff
@@ -61,17 +63,32 @@ function processPersistedRequestsQueue() {
 
     const tasks = _.map(persistedRequests, request => processRequest(request)
         .then((response) => {
-            if (response.jsonCode !== CONST.HTTP_STATUS_CODE.SUCCESS) {
-                throw new Error(`Persisted request failed due to jsonCode: ${response.jsonCode}`);
+            if (response.jsonCode === CONST.JSON_CODE.NOT_AUTHENTICATED) {
+                getLogger().info('Persisted optimistic request needs authentication');
+            } else {
+                getLogger().info('Persisted optimistic request returned a valid jsonCode. Not retrying.');
             }
-
+            onResponse(request, response);
             NetworkRequestQueue.removeRetryableRequest(request);
         })
         .catch((error) => {
-            const retryCount = NetworkRequestQueue.incrementRetries(request);
-            getLogger().info('Persisted request failed', false, {retryCount, command: request.command, error: error.message});
-            if (retryCount >= CONST.NETWORK.MAX_REQUEST_RETRIES) {
-                // Request failed too many times removing from persisted storage
+            // If we are catching a known network error like "Failed to fetch" allow this request to be retried if we have retries left
+            if (error.message === CONST.ERROR.FAILED_TO_FETCH) {
+                const retryCount = NetworkRequestQueue.incrementRetries(request);
+                getLogger().info('Persisted request failed', false, {retryCount, command: request.command, error: error.message});
+                if (retryCount >= CONST.NETWORK.MAX_REQUEST_RETRIES) {
+                    getLogger().info('Request failed too many times removing from storage', false, {retryCount, command: request.command, error: error.message});
+                    NetworkRequestQueue.removeRetryableRequest(request);
+                }
+            } else if (error.name === CONST.ERROR.REQUEST_CANCELLED) {
+                getLogger().info('Persisted request was cancelled. Not retrying.');
+                onError(request);
+                NetworkRequestQueue.removeRetryableRequest(request);
+            } else {
+                getLogger().alert(`${CONST.ERROR.ENSURE_BUGBOT} unknown error while retrying persisted request. Not retrying.`, {
+                    command: request.command,
+                    error: error.message,
+                });
```

Comment:
NAB: I think we can also move this logic to `processRequest`, no?

Reply:
Same answer as before.